pkg/csi/manila/keyset/keyset.go

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package keyset
+
+import "sync"
+
+type KeySet struct {
```

Comment:
isn't it better to use https://golang.org/pkg/sync/atomic/#Value ?

Reply:
Can you be please more specific? Instead of what?

Reply:
What I'm confused about is that `Value` doesn't really seem to fit here. It's a single-value structure, whereas we need to store multiple keys (resource names in this case). It doesn't offer _compare-and-swap_ which is needed for checking and noting that an operation is in progress. Using it as a value inside the map wouldn't help either because the map itself must be protected against races when writing into it. `Value` wouldn't prevent from races here. Maybe you had a different idea?

Reply:
`sync.Map` and its `LoadOrStore()` seems like a much better contender. Using that instead of `keyset` is possible.